src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs

```diff
@@ -48,12 +52,20 @@ public void BeginRequest(HttpContext httpContext, HostingApplication.Context con
             var diagnosticListenerEnabled = _diagnosticListener.IsEnabled();
             var loggingEnabled = _logger.IsEnabled(LogLevel.Critical);
 
+            // TODO: If loggingEnabled check to change to if ActivityPropogation options are enabled
             if (loggingEnabled || (diagnosticListenerEnabled && _diagnosticListener.IsEnabled(ActivityName, httpContext)))
             {
-                context.Activity = StartActivity(httpContext, out var hasDiagnosticListener);
-                context.HasDiagnosticListener = hasDiagnosticListener;
+                _dummyListener = new ActivityListener()
```

Comment:
>_dummyListener [](start = 16, length = 14)

This will create a listener with every request if the logging enabled? would be good if you can cache this object first time it get used. Instead of dispose later, you may do _dummyListener.Sample = null instead.

Reply:
Hi.
I think the dummyListener (its not really dummy now!) should be created once in startup somewhere.
But its Sample callback should encapsulate the `if (loggingEnabled || (diagnosticListenerEnabled && _diagnosticListener.IsEnabled(ActivityName, httpContext)))` logic. 

So that we'll get previous behaviour. ie dummy listener will force activity creation (by returning PropagationData), when the above condition is met.

Reply:
This seems convoluted

Reply:
Agree its more complex than I was originally imagining. (which was simply replace new Activity().Start with ActivitySource.Start). 

(@shirhatti should we try an alternate PR like the one in AzureSDK and compare to see if that would be more neat)

Reply:
I am not seeing this is really complicated. the change here can moved up to cache the listener (at the place we define it) and then the change in the StartActivity is trivial. 

---
In reply to: [575399486](https://github.com/dotnet/aspnetcore/pull/30089#discussion_r575399486) [](ancestors = 575399486)

Reply:
I filed an issue based on my conversation with @tarekgh and @Tratcher - https://github.com/dotnet/runtime/issues/48237.

Reply:
Resolving as we're no longer creating a dummy listener